pkg/controller/gitopsconfig/controller.go

```diff
@@ -75,8 +76,22 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// sigs.k8s.io/controller-runtime v0.1.10 is used now
+	// If the version were at least 0.2.0, a ready predicate.GenerationChangedPredicate could be used instead of writing this one
+	// https://godoc.org/sigs.k8s.io/controller-runtime/pkg/predicate#GenerationChangedPredicate
+	generationChangedPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If there's a status update, .metadata.Generation field isn't changed - ignore such event
+			return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()
+		},
+	}
+
 	// Watch for changes to primary resource GitOpsConfig
-	err = c.Watch(&source.Kind{Type: &gitopsv1alpha1.GitOpsConfig{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(
+		&source.Kind{Type: &gitopsv1alpha1.GitOpsConfig{}},
+		&handler.EnqueueRequestForObject{},
+		generationChangedPredicate,
```

Comment:
[optional] What would you think about moving the predicate.Funcs object inline here, instead of creating an extra variable?